### PR TITLE
GsaGH-307 update import methods

### PR DESCRIPTION
**Pull request by kpne**

### CODE REVIEW

File: GsaGH/Components/1_Properties/EditMaterial.cs [hunk 1]
```diff
@@ -82,97 +82,97 @@ public class EditMaterial : GH_OasysComponent {
...
-
-              break;
-            }
-
-          default:
-            this.AddRuntimeError("Unable to convert Material Type input");
-            return;
-        }
-      }
+      //if (da.GetData(3, ref ghTyp)) {
```
Comment (DominikaLos) [inline GsaGH/Components/1_Properties/EditMaterial.cs:85]: as above

File: GsaGH/Components/GraveyardComp/CreateCustomMaterial_OBSOLETE.cs [hunk 2]
```diff
@@ -173,41 +173,41 @@ CoefficientOfThermalExpansionUnit thermalExpansionUnit
...
       };
 
-      material.GradeProperty = 0;
+      material.Id = 0;
 
-      switch (_mode) {
-        case FoldMode.Generic:
-          material.MaterialType = GsaMaterial.MatType.Generic;
-          break;
+      //switch (_mode) {
```
Comment (DominikaLos) [inline GsaGH/Components/GraveyardComp/CreateCustomMaterial_OBSOLETE.cs:178]: as above

File: GsaGH/Components/GraveyardComp/CreateMaterial2_OBSOLETE.cs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+    protected override void SolveInstance(IGH_DataAccess da) {
+      var material = new GsaMaterial();
+
+      int grade = 0;
+      if (da.GetData(0, ref grade)) {
+        material.Id = grade;
+      }
+
+      //switch (_selectedItems[0]) {
```
Comment (DominikaLos) [inline GsaGH/Components/GraveyardComp/CreateMaterial2_OBSOLETE.cs:76]: as above

File: GsaGH/Components/GraveyardComp/CreateMaterial_OBSOLETE.cs [hunk 4]
```diff
@@ -151,48 +151,48 @@ private enum FoldMode {
...
         GH_Convert.ToInt32(ghGrade, out int grade, GH_Conversion.Both);
-        material.GradeProperty = grade;
+        material.Id = grade;
       }
 
-      switch (_mode) {
-        case FoldMode.Generic:
-          material.MaterialType = GsaMaterial.MatType.Generic;
-          break;
+      //switch (_mode) {
```
Comment (DominikaLos) [inline GsaGH/Components/GraveyardComp/CreateMaterial_OBSOLETE.cs:163]: as above

File: GsaGH/Components/GraveyardComp/EditMaterial_OBSOLETE.cs [hunk 5]
```diff
@@ -67,106 +67,106 @@ public class EditMaterial_OBSOLETE : GH_OasysComponent {
...
-
-                break;
-              }
-
-            default:
-              this.AddRuntimeError("Unable to convert Material Type input");
-              return;
-          }
-        }
+        //if (da.GetData(2, ref ghTyp)) {
```
Comment (DominikaLos) [inline GsaGH/Components/GraveyardComp/EditMaterial_OBSOLETE.cs:75]: as above

File: GsaGHTests/1_BaseParameters/1_Properties/GsaProp2dTest.cs [hunk 6]
```diff
@@ -80,18 +80,18 @@ public class GsaProp2dTests {
         ReferenceSurface = referenceSurface,
         AdditionalOffsetZ = offset
       };
-      var mat = new GsaMaterial((int)materialType) {
-        AnalysisProperty = materialAnalysisProperty,
-      };
-      orig.Material = mat;
+      //var mat = new GsaMaterial((int)materialType) {
```
Comment (DominikaLos) [inline GsaGHTests/1_BaseParameters/1_Properties/GsaProp2dTest.cs:83]: as above

File: GsaGHTests/1_BaseParameters/1_Properties/GsaSectionTest.cs [hunk 7]
```diff
@@ -88,10 +88,10 @@ public class GsaSectionTest {
       double myarea1 = 7808.121;
       var orig = new GsaSection(profile) {
         MaterialId = 1,
-        Material = {
-          GradeProperty = 2,
-          MaterialType = GsaMaterial.MatType.Steel,
-        },
+        //Material = {
```
Comment (DominikaLos) [inline GsaGHTests/1_BaseParameters/1_Properties/GsaSectionTest.cs:91]: as above

**Review by DominikaLos — DISMISSED** [review]

File: GsaGH/Components/1_Properties/CreateCustomMaterial.cs [hunk 8]
```diff
@@ -175,41 +184,41 @@ CoefficientOfThermalExpansionUnit thermalExpansionUnit
          .As(CoefficientOfThermalExpansionUnit.InverseDegreeCelsius),
       };
 
-      material.GradeProperty = 0;
+      material.Id = 0;
```
[the remark this answers is not in the document]
Comment (kpne) [reply GsaGH/Components/1_Properties/CreateCustomMaterial.cs:187]: done

File: GsaGH/Components/1_Properties/EditMaterial.cs (hunk 1 above)
Comment (kpne) [reply GsaGH/Components/1_Properties/EditMaterial.cs:85]: done

File: GsaGH/Components/GraveyardComp/CreateCustomMaterial_OBSOLETE.cs (hunk 2 above)
Comment (kpne) [reply GsaGH/Components/GraveyardComp/CreateCustomMaterial_OBSOLETE.cs:178]: uncommented

File: GsaGH/Components/GraveyardComp/CreateMaterial_OBSOLETE.cs (hunk 4 above)
Comment (kpne) [reply GsaGH/Components/GraveyardComp/CreateMaterial_OBSOLETE.cs:163]: uncommented

File: GsaGH/Components/GraveyardComp/EditMaterial_OBSOLETE.cs (hunk 5 above)
Comment (kpne) [reply GsaGH/Components/GraveyardComp/EditMaterial_OBSOLETE.cs:75]: uncommented

File: GsaGH/Components/GraveyardComp/CreateMaterial2_OBSOLETE.cs (hunk 3 above)
Comment (kpne) [reply GsaGH/Components/GraveyardComp/CreateMaterial2_OBSOLETE.cs:76]: updated with error message

File: GsaGHTests/1_BaseParameters/1_Properties/GsaSectionTest.cs (hunk 7 above)
Comment (kpne) [reply GsaGHTests/1_BaseParameters/1_Properties/GsaSectionTest.cs]: updated test

File: GsaGHTests/1_BaseParameters/1_Properties/GsaProp2dTest.cs (hunk 6 above)
Comment (kpne) [reply GsaGHTests/1_BaseParameters/1_Properties/GsaProp2dTest.cs]: updated test

File: GsaGH/Components/GraveyardComp/EditMaterial_OBSOLETE.cs (hunk 5 above)
Comment (kpne) [reply GsaGH/Components/GraveyardComp/EditMaterial_OBSOLETE.cs]: updated

**Review by DominikaLos — CHANGES_REQUESTED** [review]

File: GsaGH/Components/GraveyardComp/EditMaterial_OBSOLETE.cs [hunk 9]
```diff
@@ -60,116 +60,110 @@ public class EditMaterial_OBSOLETE : GH_OasysComponent {
 
       GsaMaterialGoo materialGoo = null;
       if (da.GetData(0, ref materialGoo)) {
-        material = materialGoo.Value.Duplicate();
+        material = materialGoo.Value.Duplicate(true);
```
[the remark this answers is not in the document]
Comment (kpne) [reply GsaGH/Components/GraveyardComp/EditMaterial_OBSOLETE.cs]: https://github.com/arup-group/GSA-Grasshopper/pull/421/commits/6561e960b29bc4ac300fb1f8ac61713bd8d07cef

File: GsaGH/Components/GraveyardComp/CreateMaterial_OBSOLETE.cs [hunk 10]
```diff
@@ -146,55 +146,12 @@ private enum FoldMode {
     }
 
     protected override void SolveInstance(IGH_DataAccess da) {
-      var material = new GsaMaterial();
+      AddRuntimeMessage(GH_RuntimeMessageLevel.Error,
```
[the remark this answers is not in the document]
Comment (kpne) [reply GsaGH/Components/GraveyardComp/CreateMaterial_OBSOLETE.cs]: https://github.com/arup-group/GSA-Grasshopper/pull/421/commits/6594556cb4fc749ed8ec18d319f016e16fd8e5ce

**Review by DominikaLos — APPROVED** [review]

**Pull request merged**